lift/src/main/scala/bootstrap/liftweb/Boot.scala

```diff
@@ -22,5 +22,17 @@ class Boot {
 
     // Force the request to be UTF-8
     LiftRules.early.append(_.setCharacterEncoding("UTF-8"))
+    
+    def hasTrailingIndex_?(path: ParsePath): Boolean = path.partPath.lastOption.contains("index")

```

Comment:
I prefer the `index` solution personally, it will remove all the rest of the bad hack, let see what @mfirry thinks about :-)